Sources/JExtractSwiftLib/JNI/JNISwift2JavaGenerator+JavaBindingsPrinting.swift

```diff
@@ -176,75 +176,80 @@ extension JNISwift2JavaGenerator {
   }
 
   private func printFunctionBinding(_ printer: inout CodePrinter, _ decl: ImportedFunc) {
-    guard let _ = translatedDecl(for: decl) else {
+    guard let translatedDecl = translatedDecl(for: decl) else {
       // Failed to translate. Skip.
       return
     }
 
+    var modifiers = "public"
     if decl.isStatic || decl.isInitializer || !decl.hasParent {
-      printStaticFunctionBinding(&printer, decl)
-    } else {
-      printMemberMethodBindings(&printer, decl)
+      modifiers.append(" static")
     }
-  }
 
-  private func printStaticFunctionBinding(_ printer: inout CodePrinter, _ decl: ImportedFunc) {
-    printDeclDocumentation(&printer, decl)
-    printer.print(
-      "public static native \(renderFunctionSignature(decl));"
-    )
-  }
-
-  /// Renders Java bindings for member methods
-  ///
-  /// Member methods are generated as a function that extracts the `selfPointer`
-  /// and passes it down to another native function along with the arguments
-  /// to call the Swift implementation.
-  private func printMemberMethodBindings(_ printer: inout CodePrinter, _ decl: ImportedFunc) {
-    let translatedDecl = translatedDecl(for: decl)! // We will only call this method if we can translate the decl.
+    let translatedSignature = translatedDecl.translatedFunctionSignature
+    let resultType = translatedSignature.resultType.javaType
+    var parameters = translatedDecl.translatedFunctionSignature.parameters.map(\.parameter.asParameter)
+    if translatedSignature.requiresSwiftArena {
+      parameters.append("SwiftArena swiftArena$")
+    }
+    let throwsClause = decl.isThrowing ? " throws Exception" : ""
 
     printDeclDocumentation(&printer, decl)
-    printer.printBraceBlock("public \(renderFunctionSignature(decl))") { printer in
-      var arguments = translatedDecl.translatedFunctionSignature.parameters.map(\.name)
-
-      let selfVarName = "self$"
-      arguments.append(selfVarName)
+    printer.printBraceBlock(
+      "\(modifiers) \(resultType) \(translatedDecl.name)(\(parameters.joined(separator: ", ")))\(throwsClause)"
+    ) { printer in
+      printDowncall(&printer, decl)
+    }
 
-      let returnKeyword = translatedDecl.translatedFunctionSignature.resultType.isVoid ? "" : "return "
+    printNativeFunction(&printer, decl)
+  }
 
-      printer.print(
-        """
-        long \(selfVarName) = this.$memoryAddress();
-        \(returnKeyword)\(translatedDecl.parentName).$\(translatedDecl.name)(\(arguments.joined(separator: ", ")));
-        """
-      )
+  private func printNativeFunction(_ printer: inout CodePrinter, _ decl: ImportedFunc) {
+    let translatedDecl = translatedDecl(for: decl)! // Will always call with valid decl
+    let nativeSignature = translatedDecl.nativeFunctionSignature
+    let resultType = nativeSignature.result.javaType
+    let nativeName = "$\(translatedDecl.name)"
```

Comment:
Since we have `translatedDecl.nativeFunctionSignature` would we also want to put `translatedDecl.nativeFunctionBaseName` maybe that does the `$` prefix